model/patch/patch.go

```diff
@@ -116,51 +115,42 @@ type SyncAtEndOptions struct {
 	Timeout       time.Duration  `bson:"timeout,omitempty"`
 }
 
+type BackportInfo struct {
+	PatchID string `bson:"patch_id,omitempty" json:"patch_id,omitempty"`
+	SHA     string `bson:"sha,omitempty" json:"sha,omitempty"`
+}
+
 // Patch stores all details related to a patch request
 type Patch struct {
-	Id              mgobson.ObjectId `bson:"_id,omitempty"`
-	Description     string           `bson:"desc"`
-	Project         string           `bson:"branch"`
-	Githash         string           `bson:"githash"`
-	PatchNumber     int              `bson:"patch_number"`
-	Author          string           `bson:"author"`
-	Version         string           `bson:"version"`
-	Status          string           `bson:"status"`
-	CreateTime      time.Time        `bson:"create_time"`
-	StartTime       time.Time        `bson:"start_time"`
-	FinishTime      time.Time        `bson:"finish_time"`
-	BuildVariants   []string         `bson:"build_variants"`
-	Tasks           []string         `bson:"tasks"`
-	VariantsTasks   []VariantTasks   `bson:"variants_tasks"`
-	SyncAtEndOpts   SyncAtEndOptions `bson:"sync_at_end_opts,omitempty"`
-	Patches         []ModulePatch    `bson:"patches"`
-	Activated       bool             `bson:"activated"`
-	PatchedConfig   string           `bson:"patched_config"`
-	Alias           string           `bson:"alias"`
-	BackportOf      string           `bson:"backport_of"`
-	MergePatch      string           `bson:"merge_patch"`
-	GithubPatchData GithubPatch      `bson:"github_patch_data,omitempty"`
+	Id              mgobson.ObjectId       `bson:"_id,omitempty"`
+	Description     string                 `bson:"desc"`
+	Project         string                 `bson:"branch"`
+	Githash         string                 `bson:"githash"`
+	PatchNumber     int                    `bson:"patch_number"`
+	Author          string                 `bson:"author"`
+	Version         string                 `bson:"version"`
+	Status          string                 `bson:"status"`
+	CreateTime      time.Time              `bson:"create_time"`
+	StartTime       time.Time              `bson:"start_time"`
+	FinishTime      time.Time              `bson:"finish_time"`
+	BuildVariants   []string               `bson:"build_variants"`
+	Tasks           []string               `bson:"tasks"`
+	VariantsTasks   []VariantTasks         `bson:"variants_tasks"`
+	SyncAtEndOpts   SyncAtEndOptions       `bson:"sync_at_end_opts,omitempty"`
+	Patches         []ModulePatch          `bson:"patches"`
+	Activated       bool                   `bson:"activated"`
+	PatchedConfig   string                 `bson:"patched_config"`
+	Alias           string                 `bson:"alias"`
+	BackportOf      BackportInfo           `bson:"backport_of"`
```

Comment:
> We should make this omitempty

done.

> Also, have people been using this feature already, as in do we need to handle that old patches won't fit this data model now?

If they try it with a CLI that has the flag (post last week) and without this merge I think the patch will fail (because we try to unmarshal a string into an object), but I think it's okay to assume they'll try to update their CLI. We can include a warning to update the CLI in an email to server letting them know about the backport feature.
I chose to accept this deficiency to avoid a clumsy interface.

Reply:
But this also means that if we revert, anyone with the updated version will have the patch fail as well, right? It may be worth making a new BackportInfo field to avoid that, and remove BackportOf in the future.

Reply:
that makes sense. I restored the `BackportOf` string field and opened [a ticket](https://jira.mongodb.org/browse/EVG-12872) to remove it in the future.

Reply:
Awesome!

Also, I meant more specifically that omitempty should be for BackportOf because we'd want to omit the empty struct.

Reply:
ah, in the patch document.
done